pkg/catalog/catalog.go

```diff
@@ -991,23 +990,79 @@ func (c *Catalog) DiffUncommitted(ctx context.Context, repository string, branch
 	}
 	it := NewEntryDiffIterator(iter)
 	defer it.Close()
-	return listDiffHelper(it, limit, after)
+	return listDiffHelper(it, prefix, delimiter, limit, after)
+}
+
+// GetStartPos finds the correct place to start iterating from based on delimiter, prefix & after
+func GetStartPos(prefix, after, delimiter string) string {
+	switch {
+	case prefix == "" && after == "":
+		// if no prefix given and no after given, start at the beginning
+		return ""
+	case delimiter != "" && after != "":
+		// if we have a delimiter and after is not empty, start at the next common prefix after "after"
+		return string(graveler.UpperBoundForPrefix([]byte(after)))
+	case after != "":
+		// otherwise if after is set and no delimiter, continue from after
+		return after
+	default:
+		// whether we have a delimiter or not, if after is not set, start at prefix
+		return prefix
+	}
```

Comment:
This looks like it might be an easier `if`-chain if we reverse main comparison in the `switch`:
```suggestion
	if after == "" {
		// whether we have a delimiter or not, if after is not set, start at prefix
		return prefix
	}
	if delimiter == "" {
		// no delimiter, continue from after
		return after
	}
	// there is a delimiter and after is not empty, start at the next common prefix after "after"
		return string(graveler.UpperBoundForPrefix([]byte(after)))
```

Reply:
Done